src/pages/Dashboard/Activities/index.tsx

```diff
@@ -25,45 +40,92 @@ const Box = (): JSX.Element => (
-        startTime: data.activityStart,
-        endTime: data.activityEnd,
-      },
-      address: {
-        // TODO: Add address information to form
-        mention: data.activityLocation,
-        lineOne: '1234 Main Street',
-        lineTwo: '',
-        city: 'Montreal',
-        state: 'QC',
-        country: 'Canada',
-        postalCode: 'EXAMPLE',
+        title: data.title,
+        address: data.address,
+        startTime: data.startTime,
+        endTime: data.endTime,
+        description: data.description,
+        costPerIndividual: data.costPerIndividual,
+        costPerGroup: data.costPerGroup,
+        groupSize: data.groupSize,
       },
-      // image: data.activityImage,  // TODO: Add image information
     };
-    await addActivity(businessId ?? '', reqData);
+
+    await addActivity(businessId ?? '', reqData).then(() => {
+      setCurrentSection(Section.Table);
+    });
+  };
+
+  const handleActivityDeletion = async (
+    activityIds: string[],
+  ): Promise<void> => {
+    await deleteActivities(activityIds, businessId ?? '').then(async () => {
+      setActivities(
+        activities.filter((activity) => !activityIds.includes(activity.id)),
+      );
+    });
+  };
+
+  const ActionList = (): JSX.Element => {
+    return (
+      <div className="flex space-x-2">
+        <Button
+          text={constant.ADD_ACTIVITY_BUTTON}
+          hasIcon={true}
+          icon={IconNames.ADD}
+          iconPosition="lhs"
+          variant="outline"
+          onClick={() => {
+            if (currentSection !== Section.Add) {
+              setCurrentSection(Section.Add);
+            } else {
+              setCurrentSection(Section.Table);
+            }
+          }}
+          size="sm"
+        />
+        <Button
+          text={constant.DELETE_ACTIVITY_BUTTON}
+          hasIcon={true}
+          icon={IconNames.DELETE}
+          iconPosition="lhs"
+          variant="outline"
+          onClick={() => {
+            if (currentSection !== Section.Delete) {
+              setCurrentSection(Section.Delete);
+            } else {
+              setCurrentSection(Section.Table);
+            }
+          }}
+          size="sm"
+        />
+      </div>
+    );
   };
 
   useEffect(() => {
-    void (async function getAllEmployees() {
-      await getActivites(businessId ?? '')
+    void (async function () {
+      await getActivities(businessId ?? '')
         .then(async (res) => {
+          // @ts-expect-error
           const { activities } = res;
-          const activitiesArray = activities.map((activity) => ({
-            ...activity,
+          // @ts-expect-error
+          const activityArray = activities.map((activity) => ({
+            id: activity.id,
+            // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+            title: activity.title,
+            startTime: (activity.startTime).substring(0, 10),
+            endTime: (activity.endTime).substring(0, 10),
```

Comment:
```suggestion
            startTime: (activity.startTime)?.substring(0, 10),
            endTime: (activity.endTime)?.substring(0, 10),
```

To make sure that there is no error when the value is null